core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -187,7 +187,7 @@ class TransactionCoordinator(brokerId: Int,
     } else {
       // caller should have synchronized on txnMetadata already
       txnMetadata.state match {
-        case PrepareAbort | PrepareCommit =>
+        case PrepareAbort | PrepareCommit | Dead =>
```

Comment:
We should ensure that the `Dead` state is covered by the match. If we are sure it should not happen, then we can raise an exception.